R/teal_data-extract.R

```diff
@@ -0,0 +1,33 @@
+#'
+#' @section Subsetting:
+#' `x[names]` subsets objects in `teal_data` environment and limit the code to the necessary needed to build limited
+#' objects.
+#'
+#' @param names (`character`) names of objects included in `teal_subset` to subset
+#' @param x (`teal_data`)
+#'
+#' @examples
+#'
+#' # Subsetting
+#' data <- teal_data()
+#' data <- eval_code(data, "a <- 1;b<-2")
+#' data["a"]
+#' data[c("a", "b")]
+#'
+#' join_keys(data) <- join_keys(join_key("a", "b", "x"))
+#' join_keys(data["a"]) # should show empty keys
+#' join_keys(data["b"])
+#' join_keys(data)["a"] # should show empty keys
+#' join_keys(data)["b"]
+#'
+#' @rdname teal_data
+#'
+#' @export
+`[.teal_data` <- function(x, names) {
+  x <- NextMethod("`[`", x, check_names = FALSE) # takes 'names' from function's environment
```

Comment:
`check_names = FALSE`

because

```r
td <- teal_data(a = 1, b = 2)
```

has no code, and 

```r
td['a']
```

raises a warning that `a` is missing in the code when extracting code

Reply:
It should throw a warning. One tries to extract `a` which is not present in the code.

Reply:
Or maybe a question, WDYT about this? Should it throw or not?

Reply:
Maybe we can have an error on `teal_data` level, that warns about it but with custom warning for teal_data curated for this siu. Without `check_names = FALSE` it gives unreasonable warnning from `get_code_dependency`.

Reply:
This is passed to `get_code` -> `get_code_dependency` and it checks for objects within code
https://github.com/insightsengineering/teal.code/pull/216/files#diff-66a61facc4f5ca86215d63af40e2645b364950439051784916316ef78d5073c8R34